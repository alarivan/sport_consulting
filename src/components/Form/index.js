import React from "react"
import { Formik, Form } from "formik"

import InputGroup from "../common/InputGroup"
import Select from "../common/Select"
import Button from "../styled/Button.css.js"
import Input from "../styled/TextInput.css.js"
import Textarea from "../styled/Textarea.css.js"

import { validationSchema, options, initialValues } from "../../helper/form"

export default ({ onFormSubmit }) => {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={(values, { setSubmitting }) => {
        onFormSubmit(values, setSubmitting)
      }}
    >
      {({ isSubmitting, values, handleChange, handleBlur }) => (
        <Form
          name="contact"
          method="post"
          data-netlify="true"
          data-netlify-honeypot="bot-field"
        >
          <input type="hidden" name="form-name" value="contact" />
          <InputGroup
            label="Full Name"
            error={{ name: "name", component: "div" }}
          >
            <Input
              onChange={handleChange}
              onBlur={handleBlur}
              type="text"
              name="name"
              value={values.name}
              placeholder="Jon Doe"
            />
          </InputGroup>

          <InputGroup label="Email" error={{ name: "email", component: "div" }}>
            <Input
              onChange={handleChange}
              onBlur={handleBlur}
              type="email"
              name="email"
              value={values.email}
              placeholder="jondoe@email.com"
            />
          </InputGroup>

          <InputGroup
            label="Location"
            error={{ name: "location", component: "div" }}
          >
            <Input
              onChange={handleChange}
              onBlur={handleBlur}
              type="text"
              name="location"
              value={values.location}
              placeholder="Los Angeles, California"
            />
          </InputGroup>

          <InputGroup
            label="Phone Number"
            error={{ name: "phone", component: "div" }}
          >
            <Input
              onChange={handleChange}
              onBlur={handleBlur}
              type="text"
              name="phone"
              value={values.phone}
              placeholder="+x xxx xxx xxxx"
            />
          </InputGroup>

          <InputGroup
            label="Subject"
            error={{ name: "subject", component: "div" }}
          >
            <Select
              name="subject"
              value={values.subject}
              options={options}
              onChange={handleChange}
              onBlur={handleBlur}
            />
          </InputGroup>

          <InputGroup
            label="Additional Information"
            error={{ name: "message", component: "div" }}
          >
            <Textarea
              onChange={handleChange}
              onBlur={handleBlur}
              name="message"
              value={values.message}
              placeholder="Message"
            />
          </InputGroup>

          <Button block type="submit" disabled={isSubmitting}>
            {isSubmitting ? "Submitting..." : "Submit Request"}
          </Button>
        </Form>
      )}
    </Formik>
  )
}
