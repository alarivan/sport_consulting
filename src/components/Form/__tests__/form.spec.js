import React from "react"
import { render } from "@testing-library/react"
import Form from "../index.js"

describe("Form", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Form />)

    expect(asFragment()).toMatchSnapshot()
  })
})
