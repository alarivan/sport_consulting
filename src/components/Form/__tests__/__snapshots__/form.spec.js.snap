// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  margin-bottom: 0.5rem;
}

.c1 {
  display: block;
  font-size: 0.75rem;
  margin: 0.25rem 0;
}

.c3 {
  border: thin solid;
  display: block;
  padding: 0.5em;
  background: #fff;
  width: 100%;
}

.c5 {
  display: block;
  width: 100%;
  outline: none;
  border: none;
  border-radius: 3px;
  padding: 0.75rem 2rem;
  cursor: pointer;
  font-weight: 600;
  color: #fff;
  -webkit-transition: background-color 200ms ease-in-out;
  transition: background-color 200ms ease-in-out;
}

.c5:disabled {
  background: #b4b6b8;
  cursor: not-allowed;
}

.c2 {
  border: thin solid;
  display: block;
  padding: 0.5em;
  background: #fff;
  width: 100%;
}

.c4 {
  border: thin solid;
  display: block;
  padding: 0.5em;
  background: #fff;
  width: 100%;
  min-height: 10rem;
}

<form
    data-netlify="true"
    data-netlify-honeypot="bot-field"
    method="post"
    name="contact"
  >
    <input
      name="form-name"
      type="hidden"
      value="contact"
    />
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Full Name
      </label>
      <input
        class="c2"
        name="name"
        placeholder="Jon Doe"
        type="text"
        value=""
      />
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Email
      </label>
      <input
        class="c2"
        name="email"
        placeholder="jondoe@email.com"
        type="email"
        value=""
      />
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Location
      </label>
      <input
        class="c2"
        name="location"
        placeholder="Los Angeles, California"
        type="text"
        value=""
      />
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Phone Number
      </label>
      <input
        class="c2"
        name="phone"
        placeholder="+x xxx xxx xxxx"
        type="text"
        value=""
      />
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Subject
      </label>
      <select
        class="c3"
        name="subject"
      >
        <option
          value="info"
        >
          General Info
        </option>
        <option
          value="workshop"
        >
          Workshop
        </option>
        <option
          value="personal-training"
        >
          Personal Training
        </option>
        <option
          value="nutrition"
        >
          Nutrition
        </option>
        <option
          value="disc"
        >
          DISC â€“ Personality Test
        </option>
      </select>
    </div>
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Additional Information
      </label>
      <textarea
        class="c4"
        name="message"
        placeholder="Message"
      />
    </div>
    <button
      class="c5"
      type="submit"
    >
      Submit Request
    </button>
  </form>
</DocumentFragment>
`;
